chapters/slang.adoc

```diff
@@ -0,0 +1,744 @@
+// Copyright 2021-2024 The Khronos Group, Inc.
+// Copyright 2025 Holochip, Inc.
+// SPDX-License-Identifier: CC-BY-4.0
+
+ifndef::chapters[:chapters:]
+ifndef::images[:images: images/]
+
+[[slang-in-vulkan]]
+= Slang in Vulkan
+
+
+Vulkan does not directly consume shaders in a human-readable text format, but instead uses xref:{chapters}what_is_spirv.adoc[SPIR-V] as an intermediate representation. This opens the option to use shader languages other than e.g. GLSL, as long as they can target the Vulkan SPIR-V environment.
+
+link:https://github.com/shader-slang/slang[Slang] is a modern shading language and compiler designed to extend xref:{chapters}hlsl.adoc[HLSL] with advanced features. It's largely backward-compatible with HLSL 2018 but adds features like generics, interfaces, and reflection capabilities. Slang can target multiple backends, including SPIR-V for Vulkan.
+
+With link:https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst#unsupported-hlsl-features[a few exceptions], all Vulkan features and shader stages available with GLSL can be used with Slang too, including recent Vulkan additions like hardware accelerated ray tracing. On the other hand, Slang to SPIR-V supports Vulkan exclusive features that are not (yet) available in DirectX.
+
+
+[[educational-resources]]
+== Educational resources
+
+For Slang, the official link:https://github.com/shader-slang/slang[GitHub repository] and link:https://docs.shader-slang.org/en/latest/[documentation] are the best resources to get started.  Additionally, the best place to get help is the link:https://khr.io/slangdiscord[slang discord server].
+
+== Why Use Slang for Vulkan?
+
+There are several advantages to using Slang for Vulkan development:
+
+* *Cross-API compatibility*: Write shaders that can be used with both Vulkan and DirectX with minimal changes
+* *Familiar syntax*: Developers with HLSL experience can leverage their existing knowledge
+* *Advanced language features*: Access to modern programming constructs like generics and interfaces
+* *Industry adoption*: Growing adoption in game engines and graphics applications
+* *Tooling support*: Rich ecosystem of tools and IDE integrations
+
+== Differences Between HLSL and Slang
+
+While Slang is built on HLSL, it adds several powerful features that make shader development more efficient, maintainable, and flexible.
```

Comment:
It sound a bit odd to say that "Slang is built on HLSL". To me, it feels like Slang would itself use or be lowered to HLSL (which it can do, but that's not what you'd do with Vulkan, where Slang is directly compiled to SPIR-V). Perhaps this could instead say "is largely compatible with HLSL 2018"?